File: app/src/java/java/com/example/ama/viewer/data/repo/DBRepository.java
package com.example.ama.viewer.data.repo;

import com.example.ama.viewer.data.api.dto.GithubUserDTO;

import io.reactivex.Completable;
import io.reactivex.Observable;

public interface DBRepository {

    Observable<GithubUserDTO> getUserFromDb(String login);

    Completable saveUserToDb(GithubUserDTO user);
}

File: app/src/kotlin/java/com/example/ama/viewer/data/repo/DBRepository.kt
package com.example.ama.viewer.data.repo

import com.example.ama.viewer.data.entity.GithubUser
import io.reactivex.Completable
import io.reactivex.Observable

interface DBRepository {

    fun getUserFromDb(): Observable<GithubUser>

    fun saveUserToDb(user: GithubUser) : Completable
}